common/src/autogluon/common/utils/utils.py

```diff
@@ -23,23 +23,28 @@
 
 
 def setup_outputdir(path, warn_if_exist=True, create_dir=True, path_suffix=None):
+    is_s3_path = False
     if path:
-        assert isinstance(
-            path, (str, Path)
-        ), f"Only str and pathlib.Path types are supported for path, got {path} of type {type(path)}."
+        assert isinstance(path, (str, Path)), (
+            f"Only str and pathlib.Path types are supported for path, got {path} of type {type(path)}."
+        )
+
+        is_s3_path = str(path).lower().startswith("s3://")
+
     if path_suffix is None:
         path_suffix = ""
-    if path_suffix and path_suffix[-1] == os.path.sep:
+    if path_suffix and path_suffix[-1] == os.path.sep if not is_s3_path else "/":
```

Comment:
would be good to add a unit test for this, it is relevant for Windows OS

Reply:
Added more unit tests for this.